### PR TITLE
[NeMo-UX] Fixes to make PreemptionCallback work

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do ?

Adds IOMixin to PreemptionCallback to make it serializable

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/callbacks/preemption.py [hunk 1]
```diff
@@ -61,13 +64,15 @@ def on_train_end(self, trainer: Trainer, pl_module) -> None:
...
         if self.interrupted:
-            logging.info("Preemption detected, signaling trainer to stop")
-            trainer.should_stop = True
-
-    def on_exception(self, trainer: Trainer, pl_module, exception: BaseException) -> None:
-        if isinstance(exception, PreemptionException):
-            logging.info("Handling PreemptionException")
+            logging.info("Preemption detected, saving checkpoint and exiting")
             trainer.should_stop = True
+            if any(map(lambda callback: isinstance(callback, ModelCheckpoint), trainer.callbacks)):
```
Comment (ashors1) [inline nemo/lightning/pytorch/callbacks/preemption.py:69]: I think you can just use the [checkpoint_callback](https://github.com/Lightning-AI/pytorch-lightning/blob/d25014dbda553eabe12fb0669e7451ad7aa0c576/src/lightning/pytorch/trainer/trainer.py#L1305) property here, right?

**Review by ashors1 — DISMISSED** [review]
Looks good! Thanks!

**Review by athitten — APPROVED** [review]
LGTM, thanks!

**Comment by ShriyaPalsamudram** [comment]
Local test worked, LGTM. Approved.

**Review by ashors1 — APPROVED** [review]

**Review by ashors1 — APPROVED** [review]

**Pull request merged**